### ISSUE TITLE
Add syntax for ROW type

### ISSUE THREAD

**Issue by electrum** [prestodb/presto#4073]

**Comment by adamjshook** [comment]
Is this issue for supporting creating ROW data types in the DDL?  Something like:

CREATE TABLE foo (
	id VARCHAR,
	address ROW (street VARCHAR, city VARCHAR, state VARCHAR, zipcode BIGINT)
);

**Comment by martint** [comment]
It's for anywhere a data type can appear per the SQL spec (e.g., cast target, table schema definitions, etc).

Here are the relevant rules describing the syntax.

```
<row type>
  : ROW <row type body>
  ;

<row type body>
  : <left paren> <field definition> ( <comma> <field definition> )* <right paren>
  ;

<field definition>
  : <field name> <data type>
  ;

<field name>
  : <identifier>
  ;
```

Be aware that there are some changes in progress that may affect any work in this area:
https://github.com/facebook/presto/pull/4005
https://github.com/Teradata/presto/pull/88

**Comment by cberner** [comment]
This has been implemented in 0.148
